src/tls.c

```diff
@@ -927,7 +934,7 @@ static int connTLSWritev(connection *conn_, const struct iovec *iov, int iovcnt)
     /* The amount of all buffers is greater than NET_MAX_WRITES_PER_EVENT,
      * which is not worth doing so much memory copying to reduce system calls,
      * therefore, invoke connTLSWrite() multiple times to avoid memory copies. */
```

Comment:
@xbasel @ranshid is it ok?
```suggestion
    /* The amount of all buffers is greater than NET_MAX_WRITES_PER_EVENT,
     * which is not worth doing so much memory copying to reduce system calls,
     * therefore, invoke connTLSWrite() multiple times to avoid memory copies,
     * but in case when last write failed, first invoke of connTLSWrite() has to
     * provide at least the same amount of bytes as previous failed write, so
     * first element of iov has to be greater or equal to last failed write,
     * if not code will fall back to create buffer that provide enough bytes to
     * a single invoke of connTLSWrite(). */
```

Reply:
Maybe something like this:

```
/* In case the amount of all buffers is greater than NET_MAX_WRITES_PER_EVENT,
     * it might not worth doing so much memory copying to reduce system calls,
     * therefore, invoke connTLSWrite() multiple times to avoid memory copies.
     * However, in case when last write failed we still have to repeat sending last_failed_write_data_len
     * bytes. Because of openssl implementation we cannot repeat sending writes with length smaller than 
     * the last failed write (https://docs.openssl.org/master/man3/SSL_write) so in case the first io buffer
     * does not provide at least the same amount of bytes as previous failed write, we will have to fallback to 
     * memory copy to a static buffer before calling SSL_write  */
```